earthly/rust/Earthfile

```diff
@@ -135,114 +112,20 @@ CHECK:
     # to pass without needing to iterate excessively.
     RUN /scripts/std_checks.sh
 
+# Step we do during the `build` CI phase for all Rust programs
+BUILD:
+    COMMAND
+
+    RUN /scripts/std_build.sh
+
 # Check if the build executable, isn't a busted mess.
 SMOKE_TEST:
     COMMAND
-    ARG TARGETPLATFORM
-    ARG bin
+    ARG --required bin
 
     RUN ldd target/$TARGETARCH/release/$bin
     RUN readelf -p .comment target/$TARGETARCH/release/$bin
     RUN strip -v target/$TARGETARCH/release/$bin
 
     # ALL executables MUST have `--help` as an option.
     RUN target/$TARGETARCH/release/$bin --help
-
-
-# Set up our target toolchains, and copy our files.
-builder:
-    FROM +rust-base
-
-    DO +SETUP --toolchain=example/rust-toolchain.toml
-
-    DO +CP_SRC --src="example/*"
```

Comment:
This needs to be in the example, but make it more like cat-gateway now that its moved.